src/client/datascience/interactive-window/nativeInteractiveWindowProvider.ts

```diff
@@ -97,14 +121,23 @@ export class NativeInteractiveWindowProvider implements IInteractiveWindowProvid
         noop();
     }
 
-    protected async create(
-        resource: Resource,
-        mode: InteractiveWindowMode,
-        notebookUri: Uri | undefined
-    ): Promise<NativeInteractiveWindow> {
+    protected async create(resource: Resource, mode: InteractiveWindowMode): Promise<NativeInteractiveWindow> {
         // Ensure all our controllers are registered with VS Code
         await this.notebookControllerManager.loadNotebookControllers();
 
+        const { notebookUri } = (await this.commandManager.executeCommand(
+            'interactive.open',
+            ViewColumn.Beside
+        )) as INativeInteractiveWindow;
+        const notebookDocument = workspace.notebookDocuments.find(
+            (doc) => doc.uri.toString() === notebookUri.toString()
```

Comment:
```suggestion
            (doc) => doc === notebookUri
```

Reply:
This is comparing a Uri to a NotebookDocument? Not sure this works.